packages/flutter/lib/src/material/dialog.dart

```diff
@@ -316,9 +316,12 @@ class SimpleDialog extends StatelessWidget {
 
 class _DialogRoute<T> extends PopupRoute<T> {
   _DialogRoute({
-    this.child,
-    this.theme,
-  });
+    @required this.theme,
+    bool barrierDismissable: true,
```

Comment:
doesn't matter what the default is, this is only called from one place and that place lists the argument.